src/resources/transactions/transactionSimulation.ts

```diff
@@ -0,0 +1,146 @@
+import { createQueryKey, QueryConfig, QueryFunctionArgs } from '@/react-query';
+import { useQuery } from '@tanstack/react-query';
+import { RainbowError, logger } from '@/logger';
+import { metadataPOSTClient } from '@/graphql';
+import { TransactionErrorType, TransactionScanResultType, TransactionSimulationResult } from '@/graphql/__generated__/metadataPOST';
+import { isNil } from 'lodash';
+import { RequestData } from '@/redux/requests';
+import { ChainId } from '@/networks/types';
+
+type SimulationArgs = {
+  address: string;
+  chainId: ChainId;
+  isMessageRequest: boolean;
+  nativeCurrency: string;
+  req: any; // Replace 'any' with the correct type for 'req'
```

Comment:
should we do this or circle back to it?

Reply:
definitely a circle back type thing for this. It's gonna be a typescript joy.